storops/unity/resource/user_quota.py

```diff
@@ -0,0 +1,128 @@
+# coding=utf-8
+# Copyright (c) 2015 EMC Corporation.
+# All Rights Reserved.
+#
+#    Licensed under the Apache License, Version 2.0 (the "License"); you may
+#    not use this file except in compliance with the License. You may obtain
+#    a copy of the License at
+#
+#         http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+#    WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+#    License for the specific language governing permissions and limitations
+#    under the License.
+from __future__ import unicode_literals
+
+import logging
+
+import storops
+from storops.unity.resource import UnityResource, UnityResourceList
+from storops.unity.client import UnityClient
+
+__author__ = 'Rajendra Indukuri'
+
+log = logging.getLogger(__name__)
+
+
+class UnityUserQuota(UnityResource):
+    """
+    Support CRUD operations for user_quota
+
+    Operations supported:
+    create: Create user_quota on a file_system/tree_quota
+    modify: Modify user_quota on a file_system/tree_quota
+    """
+    @classmethod
+    def create(cls, cli, file_system_id=None, tree_quota_id=None,
+               hard_limit=None, soft_limit=None, uid=None,
+               unix_name=None, win_name=None):
+        """
+        Creates user_quota on the speficied filesystem or tree quota
+        :param file_system_id: This is an optional parameter which if provided
+            will create user_quota on the file_system specified. file_system_id
+            and tree_quota_id cannot be given together
+        :param tree_quota_id: This is an optional parameter which if provided
+            will create user_quota on the tree_quota specified. file_system_id
+            and tree_quota_id cannot be given together
+        :param hard_limit: sets hard_limit  on the file_system for the user
+            specified
+        :param soft_limit: sets soft_limit  on the file_system for the user
+            specified
+        :param uid: This is an optional parameter to specify user. Either of
+            uid/unix_name/win_name should be provided
+        :param unix_name: This is an optional parameter to specify user. Either
+            of uid/unix_name/win_name should be provided
+        :param win_name:This is an optional parameter to specify user. Either
+            of uid/unix_name/win_name should be provided
+        :return: created user quota.
+        """
+        fs_clz = storops.unity.resource.filesystem.UnityFileSystem
+        file_system = fs_clz.get(cli, file_system_id).verify()
```

Comment:
It is better to use filesystem_id, we also consider filesystem as one word in Unity code base.